### ISSUE TITLE
Facade AWS::createClient method doesn't work with Instance Role credentials

### ISSUE THREAD

**Issue by worldofswift** [aws/aws-sdk-php-laravel#148]
AWS-SDK v: 3.75.0
AWS-SDK-php-Laravel v: 3.3.0

The following code creates a client without any credentials (empty strings and nulls) if instance role is used:
```php
$client = AWS::createClient('sns');
```
We can investigate it when:
```php
$client->getCredentials()->wait();
```
Resulting credentials with it are recognized as non-expired.

It definitely is not a problem with the role or other code, because of the following works as expected:
```php
$client = new SnsClient([
            'region' => config('aws.region'),
            'version' => config('aws.version')
 ]);
```

And the config file:
```php
<?php

use Aws\Laravel\AwsServiceProvider;

return [
    'region' => env('AWS_SNS_REGION'),
    'version' => 'latest',
    'ua_append' => [
        'L5MOD/' . AwsServiceProvider::VERSION,
    ],
];
```

**Comment by howardlopez** [comment]
Can you provide the code you're using to assume the instance role?

**Comment by ranganathabrr** [comment]
Hi,
I am facing same issue, EC2 instance associated with IAM Role. Earlier version was working with same code provided at top.

The newer version is not working.

**Comment by ranganathabrr** [comment]
With out AWS access key & secrect AWS services not working(IAM Role associated)

It was working with out security credentials, I did composer update, and its updated new PHP SDK version. My AWS services stop working, If i provide security credentials its start working.

Ec2 instance associated with IAM Role, I don't want to expose security credentials.

And the config file:

<?php

use Aws\Laravel\AwsServiceProvider;

return [
    'region' => env('AWS_SNS_REGION'),
    'version' => 'latest',
    'ua_append' => [
        'L5MOD/' . AwsServiceProvider::VERSION,
    ],
];

**Comment by worldofswift** [comment]
> Can you provide the code you're using to assume the instance role?

Why do I need to use the extra code for Assume Role?
AWS SDK works out the box without any additional code.
As it said in the guide - if no credentials are provided from the previous step in Credential chain the credentials from role will be used. I just assign a role to the instance and the above code with creating SNS client directly works without STS or other services. The main idea is to Omit credentials array in the constructor of AWSClient.

Also, it works with SES driver and S3 driver for Laravel.

**Comment by howardlopez** [comment]
Ah, thanks for clarifying, I believe I misunderstood what you were saying. Looking into it.

**Comment by howardlopez** [comment]
I was able to replicate the issue and just merged in a fix in [release 3.3.1](https://github.com/aws/aws-sdk-php-laravel/releases/tag/3.3.1). Let us know if you still have any issues. Thanks again for bringing this to our attention!

**Comment by ranganathabrr** [comment]
@howardlopez Thanks it worked with the new release.
